### PR TITLE
Implementation for the Data Enclave proxy server

### PR DESCRIPTION
**Pull request by tanderegg**
Finally got LDAP working fully so the restricted area is only accessable by those in the "admin" LDAP group.

### CODE REVIEW

File: deploy/Vagrantfile [hunk 1]
```diff
@@ -88,4 +89,46 @@ Vagrant.configure(2) do |config|
...
+
+    proxy_config.vm.provision "ansible" do |ansible|
+      ansible.playbook = "deploy_proxy_server.yml"
+
+      ansible.groups = {
+        "proxy_server" => ["proxy_server"],
+        "all_groups:children" => ["proxy_server"]
+      }
+    end
+
```
Comment (exafox) [inline deploy/Vagrantfile:131]: Can we forward the port apache listens on to the host? 

`proxy_config.vm.network "forwarded_port", guest: 80, host: 8080` seems to do the trick just fine for me...

**Comment by exafox** [comment]
Can merge whenever. Builds just fine on my machine

File: deploy/Vagrantfile (hunk 1 above)
Comment (tanderegg) [inline deploy/Vagrantfile:131]: We can, although currently you can view the proxy server just by going to port 80 on 10.0.1.3 from the host.  Do we want to change ports for another reason?

**Comment by tanderegg** [comment]
Cool, go ahead and do it when you get a chance then, I'll issue a new one
when I've got the iptables stuff done, likely Monday.

File: deploy/Vagrantfile (hunk 1 above)
Comment (exafox) [inline deploy/Vagrantfile:131]: When your Dev box is a headless server (waves) you don't have a web browser that can see the private network...

**Pull request merged**